### PR TITLE
Try to use OIDC token from metadata server as first option

### PR DESCRIPTION
**Pull request by joehan**
Taking a crack at switching the Tasks code to use ID tokens from the Metadata

### CODE REVIEW

**Comment by joehan** [comment]
Apologies for the build issues on this - wanted to get something in front of you before i spent time polishing this

**Review by lahirumaramba — CHANGES_REQUESTED** [review]
Thanks @joehan ! Added a few comments.

File: src/app/credential-internal.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ const GOOGLE_AUTH_TOKEN_PATH = '/o/oauth2/token';
 // NOTE: the Google Metadata Service uses HTTP over a vlan
 const GOOGLE_METADATA_SERVICE_HOST = 'metadata.google.internal';
 const GOOGLE_METADATA_SERVICE_TOKEN_PATH = '/computeMetadata/v1/instance/service-accounts/default/token';
+const GOOGLE_METADATA_SERVICE_IDENTITY_PATH = '/computeMetadata/v1/instance/service-accounts/default/token';
```
[the remark this answers is not in the document]
Comment (joehan) [reply src/app/credential-internal.ts:35]: Ahhhh, this was a typo - it should be /identity, not /token

File: src/app/credential.ts [hunk 2]
```diff
@@ -44,4 +44,13 @@ export interface Credential {
...
   getAccessToken(): Promise<GoogleOAuthAccessToken>;
+
+    /**
+   * Returns an encoded OIDC token used to authenticate calls to 
+   * private Cloud Functions or other compute services.
+   *
+   * @param audience The URL this token will be used to call.
+   * @returns A base64 encoded OIDC token.
+   */
+  getIDToken(audience: string): Promise<string>;
```
Comment (joehan) [inline src/app/credential.ts:55]: It ok to add to this interface? or is this a public interface/would require a API review?

**Pull request closed without merging**